test/evergreen/print_python_stack_trace.py

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env python
+#
+# Public Domain 2014-present MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#
+# This is free and unencumbered software released into the public domain.
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import argparse
+import itertools
+import os
+import re
+import subprocess
+import sys
+from shutil import which
+
+
+def border_msg(msg: str):
+    count = len(msg) + 2
+    dash = "-"*count
+    return "+{dash}+\n| {msg} |\n+{dash}+".format(dash=dash, msg=msg)
+
+
+class LLDBDumper:
+    """LLDBDumper class."""
+    def __init__(self):
+        self.dbg = self.__find_debugger("lldb")
+
+    @staticmethod
+    def __find_debugger(debugger: str):
```

Comment:
I'm wondering the purpose of the double underscore here in the method name. If it's to mark an internal method as a convention, then a single underscore should be used. Double underscores have the "side effect" of naming mangling, which I guess is not the desired effect here?

Reply:
updated in [d6b8174](https://github.com/wiredtiger/wiredtiger/pull/7645/commits/d6b817445834ae25c1f1a883cb5ee7d2d3860373)
Just out of curiosity what are the side effects of name mangling?

Reply:
"side effect" may not be the appropriate description here. If you search "name mangling" in [this](https://docs.python.org/3/tutorial/classes.html) link
```Name mangling is helpful for letting subclasses override methods without breaking intraclass method calls.
```
In your case no subclass is used, but in general double underscores are used when the naming mangling feature is needed.

Reply:
makes sense, thanks!